framework/src/engine/generator/endpoint.ts

```diff
@@ -48,76 +69,84 @@ interface EndpointInit {
 export class Endpoint {
 	[key: string]: unknown;
 
-	private readonly _keypairs: dataStructures.BufferMap<Keypair>;
-	private readonly _generators: Generator[];
+	private readonly _keypairs: dataStructures.BufferMap<PlainGeneratorKeyData>;
 	private readonly _consensus: Consensus;
+	private readonly _chain: Chain;
+	private readonly _blockTime: number;
 
 	private _generatorDB!: Database;
 
 	public constructor(args: EndpointArgs) {
 		this._keypairs = args.keypair;
-		this._generators = args.generators;
 		this._consensus = args.consensus;
 		this._abi = args.abi;
+		this._chain = args.chain;
+		this._blockTime = args.blockTime;
 	}
 
 	public init(args: EndpointInit) {
 		this._generatorDB = args.generatorDB;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async getStatus(_context: RequestContext): Promise<GetStatusResponse> {
```

Comment:
I've noticed various `context` param definitions (`_context`, `ctx`, `_ctx`) in the same file/module. It's not a show stopper but should there be consistency in the naming?

Reply:
good point! `_ctx` was used when `ctx` is not used in the function so that typescript compiles, but `_context` shouldn't be used